standardization/src/main/scala/za/co/absa/enceladus/standardization/interpreter/stages/TypeParser.scala

```diff
@@ -225,20 +239,63 @@ object TypeParser {
 
   }
 
-  private trait ScalarParser extends PrimitiveParser {
+  private abstract class ScalarParser[T](implicit defaults: Defaults) extends PrimitiveParser[T] {
     override def assemblePrimitiveCastLogic: Column = column.cast(field.dataType)
   }
 
-  private trait NumericParser extends ScalarParser
+  private abstract class NumericParser[N: TypeTag](override val field: NumericTypeStructField[N])
+                                                  (implicit defaults: Defaults) extends ScalarParser[N] {
+    override def standardize(): ParseOutput = {
+      if (field.needsUdfParsing) {
+        standardizeUsingUdf()
+      } else {
+        super.standardize()
+      }
+    }
 
-  private final case class IntegralParser(field: TypedStructField,
-                                          path: String,
-                                          origSchema: StructType,
-                                          parent: Option[Parent],
-                                          overflowableTypes: Set[DataType ],
-                                          minValue: Long,
-                                          maxValue: Long
-                                         ) extends NumericParser {
+    override def assemblePrimitiveCastLogic: Column = {
+      if (origType == StringType) {
+        // in case of string as source some adjustments might be needed
+        val decimalSymbols = field.pattern.toOption.flatten.map(
+          _.decimalSymbols
+        ).getOrElse(defaults.getDecimalSymbols)
+        val replacements: Map[Char, Char] = decimalSymbols.basicSymbolsDifference(defaults.getDecimalSymbols)
+
+        val columnWithProperDecimalSymbols: Column = if (replacements.nonEmpty) {
+          val from = replacements.keys.mkString
+          val to = replacements.values.mkString
+          translate(column, from, to)
+        } else {
+          column
+        }
+
+        val columnToCast = if (field.allowInfinity && (decimalSymbols.infinityValue != InfinityStr)) {
+          // because Spark uses Java's conversion from String, which in turn checks for hardcoded "Infinity" string not
+          // DecimalFormatSymbols content
+          val infinityEscaped = Pattern.quote(decimalSymbols.infinityValue)
+          regexp_replace(regexp_replace(columnWithProperDecimalSymbols, InfinityStr, s"${InfinityStr}_"), infinityEscaped, InfinityStr)
+        } else {
+          columnWithProperDecimalSymbols
+        }
+        columnToCast.cast(field.dataType)
+      } else {
+        super.assemblePrimitiveCastLogic
+      }
+    }
+
+    private def standardizeUsingUdf(): ParseOutput = {
+      //TODO #1047 val udfFnc: UserDefinedFunction = UDFBuilder.typedStructFieldStringUdf(field, columnIdForUdf, defaultValue)
```

Comment:
Here you start with the issue number and then the comment. On some other places, you start with `TODO Desc #No` 

I would like to keep it unified and have `TODO #No Desc` if possible.